nodeinstaller/node-installer.go

```diff
@@ -91,17 +91,24 @@ func run(ctx context.Context, fetcher assetFetcher, platform platforms.Platform)
 		return fmt.Errorf("installing files: %w", err)
 	}
 
-	if err := containerdRuntimeConfig(runtimeBase, targetConf.KataConfigPath(), platform, config.QemuExtraKernelParams, config.DebugRuntime); err != nil {
+	kataRuntimeConfig, err := getContainerdRuntimeConfig(runtimeBase, platform, config.QemuExtraKernelParams, config.DebugRuntime)
+	if err != nil {
 		return fmt.Errorf("generating kata runtime configuration: %w", err)
 	}
 
-	if err := patchContainerdConfig(runtimeHandler, runtimeBase, targetConf.ContainerdConfigPath(), platform, config.DebugRuntime); err != nil {
-		return fmt.Errorf("patching containerd configuration: %w", err)
+	if err := writeContainerdRuntimeConfig(targetConf.KataConfigPath(), kataRuntimeConfig); err != nil {
+		return fmt.Errorf("writing kata runtime configuration: %w", err)
 	}
 
-	if err := installImagepullerConfig(hostMount, runtimeHandler); err != nil {
+	imagpullerConfigPath, err := installImagepullerConfig(hostMount, runtimeHandler)
+	if err != nil {
 		return fmt.Errorf("installing imagepuller config: %w", err)
 	}
+	kataRuntimeConfig.Hypervisor["qemu"]["contrast_imagepuller_config"] = imagpullerConfigPath
```

Comment:
I don't think this should be exposed at this level. Can't you just pass the `imagepullerConfigPath` into the `containerdRuntimeConfig` func and skip adding it to the kata config if it is the empty string?